From d0da12adb7bdc01bca8a685cfbe78b9e2756db26 Mon Sep 17 00:00:00 2001
From: Milos Ratkovic <milosr@gmail.com>
Date: Tue, 17 Apr 2018 01:17:09 +0200
Subject: [PATCH 54/57] Camera: Revert check metadata type before releasing
 frame

Hidl method releaseRecordingFrameHandle fails for metadata
type other than kMetadataBufferTypeNativeHandleSource which
prevents older devices to record video.
---
 .../libcameraservice/device1/CameraHardwareInterface.cpp | 16 ++++++----------
 1 file changed, 6 insertions(+), 10 deletions(-)

diff --git a/services/camera/libcameraservice/device1/CameraHardwareInterface.cpp b/services/camera/libcameraservice/device1/CameraHardwareInterface.cpp
index bdaeb14a4..89210f535 100644
--- a/services/camera/libcameraservice/device1/CameraHardwareInterface.cpp
+++ b/services/camera/libcameraservice/device1/CameraHardwareInterface.cpp
@@ -595,16 +595,12 @@ void CameraHardwareInterface::releaseRecordingFrame(const sp<IMemory>& mem)
     if (CC_LIKELY(mHidlDevice != nullptr)) {
         if (size == sizeof(VideoNativeHandleMetadata)) {
             VideoNativeHandleMetadata* md = (VideoNativeHandleMetadata*) mem->pointer();
-            if (md->eType == kMetadataBufferTypeNativeHandleSource) {
-                // Caching the handle here because md->pHandle will be subject to HAL's edit
-                native_handle_t* nh = md->pHandle;
-                hidl_handle frame = nh;
-                mHidlDevice->releaseRecordingFrameHandle(heapId, bufferIndex, frame);
-                native_handle_close(nh);
-                native_handle_delete(nh);
-            } else {
-                mHidlDevice->releaseRecordingFrame(heapId, bufferIndex);
-            }
+            // Caching the handle here because md->pHandle will be subject to HAL's edit
+            native_handle_t* nh = md->pHandle;
+            hidl_handle frame = nh;
+            mHidlDevice->releaseRecordingFrameHandle(heapId, bufferIndex, frame);
+            native_handle_close(nh);
+            native_handle_delete(nh);
         } else {
             mHidlDevice->releaseRecordingFrame(heapId, bufferIndex);
         }
-- 
2.11.0

